### DISCUSSION TITLE
How to enable CSS extraction for an environment with the output target of `node`?

### DISCUSSION THREAD

**Discussion by jdavis** [web-infra-dev/rsbuild#3898] [Q&A] [answered]
I'm not seeing it in the options but is there a way to enable CSS extraction and bundling if the output target is node? The docs say:

> CSS code will not be bundled or extracted, but the id information of CSS Modules will be included in the bundle

under this link https://rsbuild.dev/config/output/target#node-target

I tried enabling flags to trigger the `pluginCss()` but its getting stuck because that plugin only enables loaders if `target === 'web'`: https://github.com/web-infra-dev/rsbuild/blob/cea43640889fab0caeee10f9fe588b6779bbadb5/packages/core/src/plugins/css.ts#L249-L253

**Comment by chenjiahan** [comment]
Rsbuild does not yet support emitting CSS when the target is `node`, I will consider adding an option to allow this.

Can you explain when will Node.js bundles will use the emitted CSS bundles?

**Reply by jdavis** [reply]
Yeah, we have a unique situation where we need the raw CSS because we use SSR to do email rendering and then do some CSS inlining of all the CSS and also on CSS that isn't part of the bundle. It's a weird setup but also maybe there's another way to do this.

Could I also include the SSR entry into an environment with `web` and use the emitted CSS assets there? Will the classnames be the same or would this break?

**Reply by jdavis** [reply]
This might be hacky but would it be possible for me to make a plugin that is just a similar to what [css plugin](https://github.com/web-infra-dev/rsbuild/blob/cea43640889fab0caeee10f9fe588b6779bbadb5/packages/core/src/plugins/css.ts) does that I enable for the Node environment?

**Reply by chenjiahan** [reply, marked as answer]
Implementing a custom CSS plugin is not easy. I will add a new `output.emitCss` option today to allow emitting CSS bundles for Node target.

**Reply by jdavis** [reply]
@chenjiahan that's amazing, thank you so much

**Reply by jdavis** [reply]
New docs here: https://rsbuild.dev/config/output/emit-css
